pkg/flowaggregator/exporter/clickhouse.go

```diff
@@ -73,21 +74,21 @@ func (e *ClickHouseExporter) Stop() {
 }
 
 func (e *ClickHouseExporter) UpdateOptions(opt *options.Options) {
-	chInput := buildClickHouseInput(opt)
-	dsn, connect, err := clickhouseclient.PrepareClickHouseConnection(chInput)
+	chConfig := buildClickHouseConfig(opt)
+	connect, err := clickhouseclient.PrepareClickHouseConnection(chConfig)
 	if err != nil {
 		klog.ErrorS(err, "Error when checking new connection")
 		return
 	}
-	if dsn == e.chExportProcess.GetDsn() && chInput.CommitInterval == e.chExportProcess.GetCommitInterval() {
+	if reflect.DeepEqual(chConfig, e.chExportProcess.GetClickHouseConfig()) {
 		return
 	}
 	klog.InfoS("Updating ClickHouse")
-	if chInput.CommitInterval != e.chExportProcess.GetCommitInterval() {
-		e.chExportProcess.SetCommitInterval(chInput.CommitInterval)
+	if chConfig.CommitInterval != e.chExportProcess.GetCommitInterval() {
+		e.chExportProcess.SetCommitInterval(chConfig.CommitInterval)
 	}
-	if dsn != e.chExportProcess.GetDsn() {
-		e.chExportProcess.UpdateCH(dsn, connect)
+	if !reflect.DeepEqual(chConfig, e.chExportProcess.GetClickHouseConfig()) {
+		e.chExportProcess.UpdateCH(chConfig, connect)
 	}
```

Comment:
Correct me if I'm wrong: if the CommitInterval has changed, then we will call both `e.chExportProcess.SetCommitInterval` and `e.chExportProcess.UpdateCH` (because the commit interval is part of the config). The second call doesn't seem needed / desired?

Reply:
no, the second call will not be made. I had the same concern previously and I verified it will not called. And I double-checked again just now.
`if !reflect.DeepEqual(chConfig, e.chExportProcess.GetClickHouseConfig())` will return false after `e.chExportProcess.SetCommitInterval` being called

Reply:
That makes sense actually, `e.chExportProcess.GetClickHouseConfig()` will return the new value.
Thanks for confirming, next time don't hesitate to add a comment in the code :)